molpipeline/estimators/chemprop/lightning_wrapper.py

```diff
@@ -0,0 +1,181 @@
+        "devices": "auto",  # trainer._accelerator_connector._devices_flag does not really work
+        "num_nodes": trainer.num_nodes,
+        "precision": trainer.precision,
+        "logger": trainer.logger,
+        "callbacks": trainer.callbacks,  # type: ignore[attr-defined]
+        "fast_dev_run": trainer.fast_dev_run,  # type: ignore[attr-defined]
+        "max_epochs": trainer.max_epochs,
+        "min_epochs": trainer.min_epochs,
+        "max_steps": trainer.max_steps,
+        "min_steps": trainer.min_steps,
+        "max_time": max_time,
+        "limit_train_batches": trainer.limit_train_batches,
+        "limit_val_batches": trainer.limit_val_batches,
+        "limit_test_batches": trainer.limit_test_batches,
+        "limit_predict_batches": trainer.limit_predict_batches,
+        "overfit_batches": trainer.overfit_batches,  # type: ignore[attr-defined]
+        "val_check_interval": trainer.val_check_interval,
+        "check_val_every_n_epoch": trainer.check_val_every_n_epoch,
+        "num_sanity_val_steps": trainer.num_sanity_val_steps,
+        "log_every_n_steps": trainer.log_every_n_steps,  # type: ignore[attr-defined]
+        "enable_checkpointing": bool(trainer.checkpoint_callbacks),
+        "enable_progress_bar": get_enable_progress_bar(trainer),
+        "enable_model_summary": enable_progress_model_summary,
+        "accumulate_grad_batches": trainer.accumulate_grad_batches,
+        "gradient_clip_val": trainer.gradient_clip_val,
+        "gradient_clip_algorithm": trainer.gradient_clip_algorithm,
+        "deterministic": torch.are_deterministic_algorithms_enabled(),
+        "benchmark": torch.backends.cudnn.benchmark,
+        "inference_mode": trainer.predict_loop.inference_mode,
+        "use_distributed_sampler": trainer._accelerator_connector.use_distributed_sampler,  # pylint: disable=protected-access
+        "profiler": trainer.profiler,  # type: ignore[attr-defined]
+        "detect_anomaly": trainer._detect_anomaly,  # pylint: disable=protected-access
+        "barebones": trainer.barebones,
+        # "plugins": trainer.plugins,  # can not be exctracted
+        # "sync_batchnorm": trainer._accelerator_connector.sync_batchnorm,  # plugin related
+        "reload_dataloaders_every_n_epochs": trainer.reload_dataloaders_every_n_epochs,  # type: ignore[attr-defined]
+        "default_root_dir": trainer.default_root_dir,
+    }
+    return trainer_dict
+
+
+def get_non_default_params_trainer(trainer: pl.Trainer) -> dict[str, Any]:
+    """Get the parameters of the lightning trainer which are not default.
+
+    Parameters
+    ----------
+    trainer : pl.Trainer
+        The lightning trainer.
+
+    Returns
+    -------
+    dict[str, Any]
+        The parameters of the lightning trainer.
+    """
+    trainer_dict = get_params_trainer(trainer)
+    non_default_values = {}
+    for key, value in trainer_dict.items():
+        if value == TRAINER_DEFAULT_PARAMS.get(key, None):
```

Comment:
Make two checks. if key in TRAINER_DEFAULT_PARAMS and value == TRAINER_DEFAULT_PARAMS[key]

Reply:
Done